crates/core/src/factories/core_publish.rs

```diff
@@ -0,0 +1,246 @@
+
+        // Create a queue to process incoming op publishes. Incoming publishes are added to the
+        // fetch queue
+        let (incoming_publish_ops_tx, incoming_publish_ops_rx) =
+            channel::<IncomingPublishOps>(16_384);
+
+        let mut tasks = Vec::new();
+
+        // Spawn outgoing publish ops task.
+        let outgoing_publish_ops_task =
+            tokio::task::spawn(CorePublish::outgoing_publish_ops_task(
+                outgoing_publish_ops_rx,
+                space_id.clone(),
+                transport.clone(),
+            ));
+        tasks.push(outgoing_publish_ops_task);
+
+        // Spawn incoming publish ops task.
+        let incoming_publish_ops_task =
+            tokio::task::spawn(CorePublish::incoming_publish_ops_task(
+                incoming_publish_ops_rx,
+                fetch,
+            ));
+        tasks.push(incoming_publish_ops_task);
+
+        // Register transport module handler for incoming op and agent publishes.
+        let message_handler = Arc::new(PublishMessageHandler {
+            incoming_publish_ops_tx,
+        });
+
+        transport.register_module_handler(
+            space_id.clone(),
+            PUBLISH_MOD_NAME.to_string(),
+            message_handler.clone(),
+        );
+
+        Self {
+            outgoing_publish_ops_tx,
+            tasks,
+        }
+    }
+
+    async fn outgoing_publish_ops_task(
+        mut outgoing_publish_ops_rx: Receiver<OutgoingPublishOps>,
+        space_id: SpaceId,
+        transport: DynTransport,
+    ) {
+        while let Some((op_id, peer_url)) = outgoing_publish_ops_rx.recv().await
+        {
+            // Send ops publish message to peer.
+            let data = serialize_publish_ops_message(vec![op_id.clone()]);
+            if let Err(err) = transport
+                .send_module(
+                    peer_url.clone(),
+                    space_id.clone(),
+                    PUBLISH_MOD_NAME.to_string(),
+                    data,
+                )
+                .await
+            {
+                tracing::warn!(
+                    ?op_id,
+                    ?peer_url,
+                    "could not send publish ops: {err}"
+                );
+            }
+        }
+    }
+
+    async fn incoming_publish_ops_task(
+        mut response_rx: Receiver<IncomingPublishOps>,
+        fetch: DynFetch,
+    ) {
+        while let Some((op_ids, peer)) = response_rx.recv().await {
+            tracing::debug!(?peer, ?op_ids, "incoming publish ops");
+
+            // Retrieve ops to send from store.
+            if let Err(err) = fetch.request_ops(op_ids.clone(), peer).await {
+                tracing::warn!(
+                        "could not insert publish ops request into fetch queue: {err}"
+                    );
+                continue;
+            };
+        }
+    }
+}
+
+impl Drop for CorePublish {
+    fn drop(&mut self) {
+        for t in self.tasks.iter() {
+            t.abort();
```

Comment:
That's how the fetch module does it too, but I've seen that in many places it's just an `AbortHandle`. Nothing else of the task handle is used, so you could adopt that pattern.

Reply:
Changed that now to only store the `AbortHandle`s